classes/plugins/PluginGalleryDAO.inc.php

```diff
@@ -54,23 +67,79 @@ public function getNewestCompatible($application, $category = null, $search = nu
                 $plugins[$index] = $plugin;
             }
         }
+
         return $plugins;
     }
 
     /**
-     * Get the DOM document for the plugin gallery.
+     * Get the external Plugin XML document
      *
-     * @return DOMDocument
+     * @return ?string
      */
-    private function _getDocument()
+    protected function getExternalDocument(): ?string
     {
-        $doc = new DOMDocument('1.0');
         $application = Application::get();
         $client = $application->getHttpClient();
         $versionDao = DAORegistry::getDAO('VersionDAO');
         $currentVersion = $versionDao->getCurrentVersion();
-        $response = $client->request('GET', PLUGIN_GALLERY_XML_URL, ['query' => ['application' => $application->getName(), 'version' => $currentVersion->getVersionString()]]);
-        $doc->loadXML($response->getBody());
+        try {
+            $response = $client->request(
+                'GET',
+                PLUGIN_GALLERY_XML_URL,
+                [
+                    'query' => [
+                        'application' => $application->getName(),
+                        'version' => $currentVersion->getVersionString()
+                    ],
+                    'timeout' => self::DEFAULT_TIMEOUT,
+                ]
+            );
+
+            return $response->getBody();
+        } catch (Throwable $e) {
+            error_log($e->getMessage());
+
+            return null;
+        }
+    }
+
+    /**
+     * Get the cached Plugin XML document
+     *
+     * @return ?string
+     */
+    protected function getCachedDocument(): ?string
+    {
+        $cacheManager = CacheManager::getManager();
+        $cache = $cacheManager->getCache(
+            'loadPluginsXML',
+            Application::CONTEXT_SITE,
+            function (FileCache $cache) {
+                $cache->setEntireCache($this->getExternalDocument());
+            }
+        );
+
+        $cacheTime = $cache->getCacheTime();
+
+        // Checking if the cache is older than 1 day, or its null
+        if ($cacheTime === null || (time() - $cacheTime > 24 * 3600)) {
```

Comment:
`24 * 3600` could probably stand to be a class constant too.

Reply:
Indeed.